font-codegen/src/parsing.rs

```diff
@@ -246,7 +246,26 @@ pub(crate) struct FieldReadArgs {
 #[derive(Clone, Debug)]
 pub(crate) enum Condition {
     SinceVersion(VersionSpec),
-    IfFlag { field: syn::Ident, flag: syn::Path },
+    IfFlag {
+        field: syn::Ident,
+        flag: syn::Path,
+    },
+    IfCond {
+        xform: IfTransform,
+        args: Vec<IfArg>,
+    },
+}
+
+#[derive(Clone, Debug)]
+pub(crate) enum IfArg {
+    Field(syn::Ident),
+    Path(syn::Path),
+}
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) enum IfTransform {
+    FlagAndNonZero,
+    AnyFlag,
```

Comment:
somewhere in here we should be documenting what this is all used for, since it's getting a bit obscure.